packages/core/src/state/reducer.ts

```diff
@@ -20,155 +22,168 @@ export const initialState: State<unknown> = {
   lastReset: -Infinity,
 };
 
-export default function reducer(
```

Comment:
this diff view is very misleading since we added an indentation. the only actual change is the addition of these lines:

```ts
        if (action.endpoint?.optimisticUpdater) {
          optimisticResponse = action.endpoint?.optimisticUpdater.call(
            action.endpoint,
            new Snapshot(state, controller),
            // if endpoint exists, so must args; TODO: fix typing
            ...(action.meta.args as any[]),
          );
        }
```